packages/mobile/src/components/publish-content-drawer/PublishContentDrawer.tsx

```diff
@@ -71,13 +76,41 @@ const useStyles = makeStyles(({ palette, spacing }) => ({
   }
 }))
 
-export const PublishPlaylistDrawer = () => {
+export const PublishContentDrawer = () => {
   const dispatch = useDispatch()
-  const playlistId = useSelector(getPlaylistId)
   const neutral = useColor('neutral')
   const { toast } = useManualToast()
   const styles = useStyles()
-  const { isOpen, onClose } = useDrawerState('PublishPlaylistConfirmation')
+  const {
+    isOpen,
+    onClose,
+    data: { contentId, contentType }
+  } = usePublishContentModal()
+  const [dismissToastKey, setDismissToastKey] = useState<string | undefined>()
+  const currentTrack = useSelector((state) =>
+    getTrack(state, { id: contentId })
+  )
+
+  const previousIsPublishingTrack = usePrevious(currentTrack?._is_publishing)
+
+  // (Tracks only) - Check for changes to is_publishing to dismiss the toast
+  useEffect(() => {
+    if (
+      dismissToastKey &&
+      contentType === 'track' &&
+      previousIsPublishingTrack === true &&
+      currentTrack?._is_publishing === false
+    ) {
+      dispatch(manualClearToast({ key: dismissToastKey }))
+    }
+  }, [
+    contentType,
+    currentTrack,
+    currentTrack?._is_publishing,
+    dismissToastKey,
+    dispatch,
+    previousIsPublishingTrack
+  ])
```

Comment:
Rather than create a new saga to manage the side effect of dismissing the toast, I decided to just borrow the logic on web that checks `_is_publishing` and manage it here